src/hatch/project/core.py

```diff
@@ -182,41 +182,47 @@ def get_environment(self, env_name: str | None = None) -> EnvironmentInterface:
 
     # Ensure that this method is clearly written since it is
     # used for documenting the life cycle of environments.
```

Comment:
Since this is public facing documentation, what do you think about a context manager that would be used specifically when required that removes the environment upon an error?

Reply:
Moved to a context manager for this.